public/keycloak/keycloak.ts

```diff
@@ -208,15 +221,16 @@ class KeyCloakService {
   }
 
   scheduleRefreshTimer (refreshEarlyTime = 0) {
-    let refreshEarlyTimeinMilliseconds = Math.max(this.REFRESH_ATTEMPT_INTERVAL, refreshEarlyTime) * 1000
+    const refreshEarlyTimeinMilliseconds = Math.max(this.REFRESH_ATTEMPT_INTERVAL, refreshEarlyTime) * 1000
     this.scheduleRefreshToken(refreshEarlyTimeinMilliseconds)
   }
 
   private scheduleRefreshToken (refreshEarlyTimeinMilliseconds: number) {
     let refreshTokenExpiresIn = -1
     // check if refresh token is still valid . Or else clear all timers and throw errors
     if (this.kc && this.kc.timeSkew !== undefined && this.kc.refreshTokenParsed) {
-      refreshTokenExpiresIn = this.kc.refreshTokenParsed['exp']! - Math.ceil(new Date().getTime() / 1000) + this.kc.timeSkew
+      refreshTokenExpiresIn = this.kc.refreshTokenParsed['exp']! - Math.ceil(new Date().getTime() / 1000) +
```

Comment:
GitHub Actions warns about this line, and I'm not sure what the  `!` is on the first operand. Could you simplify this statement?

Reply:
The ! was to assert that this.kc.refreshTokenParsed['exp'] will never be null...eslint doesn't seem to like it though, so i've given it a default value of 0 in the situation that this.kc.refreshTokenParsed['exp'] is null

Reply:
Ah, same as `@Getter getConversionType!: EntityType` then.

I prefer the explicit default value, thanks :)